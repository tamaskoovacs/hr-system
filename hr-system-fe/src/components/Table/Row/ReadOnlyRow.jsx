import React from "react";
import './row.scss'

const ReadOnlyRow = ({ row, headers, handleEditClick, handleDeleteClick }) => {
  return (
    <>
      {Object?.entries(row)?.map(([key, value], index) => <td key={index} data-label={headers?.[index]}>{value}</td>)}
      <td>
        <button type="button" onClick={() => handleEditClick(row)}> Edit </button>
        <button type="button" onClick={() => handleDeleteClick(row?.id)}> Delete </button>
      </td>
    </>
  );
};

export default ReadOnlyRow;
