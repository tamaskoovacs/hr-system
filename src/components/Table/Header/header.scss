th {
  padding: 0.625em;
  text-align: center;

  font-size: 0.85em;
  letter-spacing: 0.1em;
  text-transform: uppercase;
}

@media screen and (max-width: 800px) {
  thead {
    border: none;
    clip: rect(0 0 0 0);
    height: 1px;
    margin: -1px;
    overflow: hidden;
    padding: 0;
    position: absolute;
    width: 1px;
  }
}
