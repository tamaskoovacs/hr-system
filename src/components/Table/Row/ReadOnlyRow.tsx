import { ReadOnlyRowProps } from 'interfaces/interfaces';
import React from 'react';
import './row.scss';

const ReadOnlyRow: React.FC<ReadOnlyRowProps> = ({ row, headers, handleEditClick, handleDeleteClick }) => {
  return (
    <>
      {Object?.entries(row)?.map(([key, value], index) => (
        <td key={index} data-label={headers?.[index]}>
          {value}
        </td>
      ))}
      <td>
        <button type="button" data-testid={`read-only-row-edit-bt-${row.id}`} onClick={() => handleEditClick(row)}>
          {'Edit'}
        </button>
        <button type="button" data-testid={`read-only-row-delete-bt-${row.id}`} onClick={() => handleDeleteClick(row?.id)}>
          {'Delete'}
        </button>
      </td>
    </>
  );
};

export default ReadOnlyRow;
