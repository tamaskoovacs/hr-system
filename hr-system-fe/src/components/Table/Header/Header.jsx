import './header.scss'

const Header = ({ headers }) => {
  return (
    <thead>
      <tr>
        {headers?.map((header) => (<th key={header}>{header}</th>))}
      </tr>
    </thead>
  );
};

export default Header;
