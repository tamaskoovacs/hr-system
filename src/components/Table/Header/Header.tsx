import { HeaderProps } from 'interfaces/interfaces';
import './header.scss';

const Header: React.FC<HeaderProps> = ({ headers }) => {
  return (
    <thead>
      <tr>
        {headers?.map((header) => (
          <th key={header}>{header}</th>
        ))}
      </tr>
    </thead>
  );
};

export default Header;
