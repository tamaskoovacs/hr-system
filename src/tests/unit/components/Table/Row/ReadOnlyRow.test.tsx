import React from 'react';
import { render } from '@testing-library/react';
import ReadOnlyRow from 'components/Table/Row/ReadOnlyRow';
import { headers } from 'tests/unit/components/Table/Header/Header.test';

export const tableRow = {
  id: 1,
  first_name: 'Ewen',
  last_name: null,
  email: 'emclewd1@bbb.org',
  date_of_birth: '15/12/1991',
  industry: 'Telecommunications Equipment',
  salary: 144392.9,
  years_of_experience: 2.8
};

describe('ReadOnlyRow component', () => {
  test('It should matches snapshot', () => {
    const view = render(
      <ReadOnlyRow row={tableRow} headers={headers} handleEditClick={jest.fn()} handleDeleteClick={jest.fn()} />
    );
    expect(view.container).toMatchSnapshot();
  });
});
