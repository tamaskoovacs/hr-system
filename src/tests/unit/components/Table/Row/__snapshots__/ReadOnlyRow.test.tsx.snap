// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component It should matches snapshot 1`] = `
<div>
  <thead>
    <tr>
      <th>
        Id
      </th>
      <th>
        First Name
      </th>
      <th>
        Last Name
      </th>
      <th>
        Email
      </th>
      <th>
        Date of birth
      </th>
      <th>
        Industry
      </th>
      <th>
        Salary
      </th>
      <th>
        Years of experience
      </th>
      <th>
        Actions
      </th>
    </tr>
  </thead>
</div>
`;

exports[`ReadOnlyRow component It should matches snapshot 1`] = `
<div>
  <td
    data-label="Id"
  >
    1
  </td>
  <td
    data-label="First Name"
  >
    Ewen
  </td>
  <td
    data-label="Last Name"
  />
  <td
    data-label="Email"
  >
    emclewd1@bbb.org
  </td>
  <td
    data-label="Date of birth"
  >
    15/12/1991
  </td>
  <td
    data-label="Industry"
  >
    Telecommunications Equipment
  </td>
  <td
    data-label="Salary"
  >
    144392.9
  </td>
  <td
    data-label="Years of experience"
  >
    2.8
  </td>
  <td>
    <button
      data-testid="read-only-row-edit-bt-1"
      type="button"
    >
      Edit
    </button>
    <button
      data-testid="read-only-row-delete-bt-1"
      type="button"
    >
      Delete
    </button>
  </td>
</div>
`;
